docs/API.md

```diff
@@ -1277,44 +1277,58 @@ Tags tags = minioClient.getObjectTags(
 Gets URL of an object useful when this object has public read access.
 
  __Parameters__
+
 | Parameter      | Type     | Description                |
 |:---------------|:---------|:---------------------------|
 | ``bucketName`` | _String_ | Name of the bucket.        |
 | ``objectName`` | _String_ | Object name in the bucket. |
 
+
 | Returns                |
 |:-----------------------|
 | _String_ - URL string. |
 
  __Example__
  ```java
-String url = minioClient.getObjectUrl("my-bucketname", "my-objectname");
+// Get URL of an object useful when this object has public read access.
+String url =
+    minioClient.getObjectUrl("my-bucketname","my-objectname");
 System.out.println("my-bucketname/my-objectname can be downloaded by " + url);
 ```
 
  <a name="getPresignedObjectUrl"></a>
-### getPresignedObjectUrl(Method method, String bucketName, String objectName, Integer expires, Map<String,String> reqParams)
-`public String getPresignedObjectUrl(Method method, String bucketName, String objectName, Integer expires, Map<String,String> reqParams)` _[[Javadoc]](http://minio.github.io/minio-java/io/minio/MinioClient.html#getPresignedObjectUrl-io.minio.http.Method-java.lang.String-java.lang.String-java.lang.Integer-java.util.Map-)_
+### getPresignedObjectUrl(GetPresignedObjectUrlArgs args)
+`public String getPresignedObjectUrl(GetPresignedObjectUrlArgs args)` _[[Javadoc]](http://minio.github.io/minio-java/io/minio/MinioClient.html#getPresignedObjectUrl-io.minio.GetPresignedObjectUrlArgs-)_
 
 Gets presigned URL of an object for HTTP method, expiry time and custom request parameters.
 
  __Parameters__
-| Parameter      | Type                 | Description                            |
-|:---------------|:---------------------|:---------------------------------------|
-| ``method``     | _[Method]_           | HTTP method to generate presigned URL. |
-| ``bucketName`` | _String_             | Name of the bucket.                    |
-| ``objectName`` | _String_             | Object name in the bucket.             |
-| ``expires``    | _Integer_            | Expiry in seconds; defaults to 7 days. |
-| ``reqParams``  | _Map<String,String>_ | Request parameters to override.        |
+ 
+| Parameter   | Type                           | Description  |
+|:------------|:-------------------------------|:-------------|
+| ``args``    | _[GetPresignedObjectUrlArgs]_  | Arguments.   |
+
 
 | Returns                |
 |:-----------------------|
 | _String_ - URL string. |
 
  __Example__
  ```java
-String url = minioClient.getPresignedObjectUrl(Method.DELETE, "my-bucketname", "my-objectname", 24 * 60 * 60,
-    reqParams);
+// Get presigned URL of an object for HTTP method, expiry time and custom request parameters.
```

Comment:
1. We need examples for `GET`/`PUT` to cover deprecated methods and `DELETE`to this method.
2. Have this examples sync in `MinioClient.java`, exampels and here.
3. Extra headers/query params should be `Multimap`